code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -527,7 +527,15 @@
 		name = "drinking glass"
 		desc = "Your standard drinking glass."
 		return
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/pen))
+		var/txt = stripped_input(user, "What would you like to name this drink?", "Mixology", "", 30)
+		if(txt)
+			user << "<span class='notice'>You rename The [src.name] to The [txt]. Fancy!</span>"
+			name = txt
+			desc = "This drink is called [txt]. Wonder what's in it..."
 
+..()
```

Comment:
When you merge this proc with the other one, this `..()` is already placed at the end, so don't merge this.